source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -199,6 +202,34 @@ void HeaderToMetadataFilter::writeHeaderToMetadata(Http::HeaderMap& headers,
   }
 }
 
+const Config* HeaderToMetadataFilter::getRouteConfig() const {
+  if (!decoder_callbacks_->route() || !decoder_callbacks_->route()->routeEntry()) {
+    return nullptr;
+  }
+
+  const auto* entry = decoder_callbacks_->route()->routeEntry();
+  const auto* per_filter_config =
+      entry->virtualHost().perFilterConfig(HttpFilterNames::get().HeaderToMetadata);
+
+  return !per_filter_config ? nullptr : dynamic_cast<const Config*>(per_filter_config);
```

Comment:
i dont think you need this conditional, dynamic_cast of nullptr is just nullptr I believe